mypy/subtypes.py

```diff
@@ -463,10 +463,15 @@ def visit_instance(self, left: Instance) -> bool:
                         assert unpacked.type.fullname == "builtins.tuple"
                         if isinstance(get_proper_type(unpacked.args[0]), AnyType):
                             return not self.proper_subtype
-                if mapped.type.fullname == "builtins.tuple" and isinstance(
-                    get_proper_type(mapped.args[0]), AnyType
-                ):
+                if all(isinstance(get_proper_type(a), AnyType) for a in mapped.args):
```

Comment:
I think this should actually be combined with the `if` just above. I just realised that the `if` I added is incomplete, it fact if all type argument are either `Any` or `*tuple[Any, ...]` we are still good. But currently we are only considering only plain `*tuple[Any, ...]` or e.g. `Any, Any, Any`, while missing `Any, *tuple[Any, ...], Any`, which should be equally fine (note you don't need to check number of unpacks, it is already validated during semantic analyzis).

Note this also equally applies to the equivalent special case for plain instances I added around line 538. You may fix that as well if you want while you are at it.